qiskit/circuit/controlflow/switch_case.py

```diff
@@ -294,13 +298,23 @@ class SwitchContext:
     def __init__(
         self,
         circuit: QuantumCircuit,
-        target: Union[Clbit, ClassicalRegister],
+        target: Clbit | ClassicalRegister | expr.Expr,
         *,
         in_loop: bool,
         label: Optional[str] = None,
     ):
         self.circuit = circuit
-        self.target = target
+        self._target = target
+        if isinstance(target, Clbit):
```

Comment:
Not really a comment to be addressed in this PR, but did we consider just always lifting the conditions and targets to `Expr`? I suppose that'd be a breaking change since then `self.{target,condition}` would be an `Expr` rather than what the user provided. But it'd at least remove this series of checks which we currently require all over the place!

Reply:
I did consider it, but wanted to minimise potential API breakage for the old paths, since the intent is likely to deprecate/remove the bare form.